src/seedu/addressbook/commands/Command.java

```diff
@@ -63,6 +64,10 @@ protected ReadOnlyPerson getTargetPerson() throws IndexOutOfBoundsException {
         return relevantPersons.get(getTargetIndex() - DISPLAYED_INDEX_OFFSET);
     }
 
+    public boolean isMutating() {
```

Comment:
It is not good for `isMutating` in Command to have a default implementation.

Having a default implementation for the `isMutating` method may not be intuitive for a class like `Command` as it is abstract in context. Why return false and not true?

Perhaps a better implementation would be make this method abstract and force all classes extended from `Command` to implement their own method definition.

There also should be comments for `isMutating()`. If not, users of the method as well as those who overrides method will not know how exactly the method behaves =) .